### ISSUE TITLE
Cannot serialize F# dictionary

### ISSUE THREAD

**Issue by xperiandri** [dotnet/corefx#40949]
``` F#
let parameters = dict <| seq { ("Secret", "secret") }
let contentJson = JsonSerializer.Serialize parameters
```

Throws `NotSupportedException` https://github.com/dotnet/corefx/blob/da1f8120aeec867bbd81829358ed348d323b1d14/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleDictionary.cs#L144

Why can it happen if parameters variable is of type `IDictionary<string,string>`?

**Comment by xperiandri** [comment]
While this `let parameters = new Dictionary<string,string> (dict <| seq { ("Secret", "secret") } )` works fine.
Looks like there is some bug in logic.

**Comment by scalablecory** [comment]
I believe `dict` creates an `IDictionary<>`, so there may be some difference there.

**Comment by ahsonkhan** [comment]
cc @layomia

**Comment by eiriktsarpalis** [comment]
> I believe dict creates an IDictionary<>, so there may be some difference there.

That's correct, that particular operator returns [a custom IDictionary implementation](https://github.com/dotnet/fsharp/blob/d3d5e30852454146876dc3af1f3e6ff21fdd67da/src/fsharp/FSharp.Core/fslib-extra-pervasives.fs#L168). I'd recommend closing this issue, since serializing arbitrary interface implementations should not be possible (that's a job for BinaryFormatter :P)
